src/MinkowskiSum.jl

```diff
@@ -343,10 +368,65 @@ function σ(d::AbstractVector{<:Real}, cms::CacheMinkowskiSum)::Vector{<:Real}
         # first-time computation of support vector
         svec = _σ_helper(d, arr)
     end
-    cache[d] = (l, svec)
+    cache[d] = CachedPair(l, svec)
     return svec
 end
 
+"""
+    forget_sets!(cms::CacheMinkowskiSum)::Int
+
+Tell a caching Minkowski sum to forget the stored sets.
+Only those sets are forgotten such that for each cached direction the support
+vector has been computed before.
+
+### Input
+
+- `cms` -- caching Minkowski sum
+
+# Output
+
+The number of sets that have been forgotten.
+
+### Notes
+
+This function should only be used under the assertion that no new directions are
+queried in the future; otherwise such support vector results will be incorrect.
+
+This implementation is optimistic and first tries to remove all sets.
+However, it verifies that for all cached directions the support vector has been
+computed before.
```

Comment:
i don't quite follow this sentence. isn't it necessary to have computed a support vector, in order to cache a direction? is there a use case where there is a cached direction for which the support vector has *not* been computed?

Reply:
Okay, okay, this is not clearly explained :smirk_cat:
You can add two sets to the sum, then compute the support vector, and then add another three sets to the sum. Now the cache only contains the vector for the first two sets.
The function `forget_sets!` wants to remove all five sets, but then detects that there is at least one direction which is only cached for the first two sets. Thus it only "forgets" the first two sets.

Reply:
Ah, got it now. Sounds good.